crafting-dead-core/src/main/java/com/craftingdead/core/capability/gun/DefaultGun.java

```diff
@@ -377,6 +370,22 @@ private void tryShoot(ILiving<?, ?> living, ItemStack itemStack) {
         shootSound = this.gunItem.getSilencedShootSound().orElse(shootSound);
       }
 
+      if(this instanceof AimableGun && ((AimableGun) this).isAiming(entity, new ItemStack(gunItem))

```

Comment:
Try and do this without the instance check, perhaps add a method to `DefaultGun` that gets called by `tryShoot` to process the shot and then override that method from `AimableGun`.